Userland/Libraries/LibC/getsubopt.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/ScopeGuard.h>
+#include <AK/StringView.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+// https://pubs.opengroup.org/onlinepubs/9699919799/functions/getsubopt.html
+int getsubopt(char** option_array, char* const* tokens, char** option_value)
+{
+    if (**option_array == '\0')
+        return -1;
+
+    auto option_string = StringView(*option_array);
+
+    auto possible_comma_location = option_string.find(',');
+    char* possible_comma_location_ptr_or_end_array = const_cast<char*>(option_string.characters_without_null_termination()) + option_string.length();
+    if (possible_comma_location.has_value()) {
+        possible_comma_location_ptr_or_end_array = const_cast<char*>(option_string.characters_without_null_termination()) + possible_comma_location.value();
+    }
```

Comment:
`Optional` does have a `value_or()` method, so I assume this could just be

```
const_cast<char*>(option_string.characters_without_null_termination()) + possible_comma_location.value_or(option_string.length());
```

Also, the name of the variable that we are storing the result in looks a bit over-the-top. How about `option_end` or something like that instead?